### ISSUE TITLE
Can't mutate nodepool count

### ISSUE THREAD

**Issue by georgiosd** [pulumi/pulumi-google-native#263]
Hey all!

I switched to google-native vs the gcp package because I wanted to set the GCFS setting which is not available in gcp package.

But, I am running into a problem because I cannot control the node pool count - there's only an initial node count setting and I have auto scaling disabled.

I tried changing the initial node count and setting the autoscaling config to false and min=max=count but no update was detected.

How do I do this?

**Comment by mikhailshilkov** [comment]
@viveklak Do you know/remember this?

**Comment by viveklak** [comment]
I haven't tried this myself but I imagine this is a side effect of how the default node pool is handled. I recommend creating a separate nodepool instead which should make it a lot more predictable. This is another reason I think we should invest in #123

**Comment by georgiosd** [comment]
I could be wrong, I'm very new to this, but AFAIK im not creating a default pool. This is my config:

```
var cluster = new Cluster(name, new()
        {
            Name = fullName,
            Location = Pulumi.GoogleNative.Config.Region,
            ReleaseChannel = new ReleaseChannelArgs
            {
                Channel = ReleaseChannelChannel.Regular
            },
            ShieldedNodes = new ShieldedNodesArgs
            {
                Enabled = true
            },
            NetworkConfig = new NetworkConfigArgs
            {
                DatapathProvider = NetworkConfigDatapathProvider.AdvancedDatapath
            },
            WorkloadIdentityConfig = new WorkloadIdentityConfigArgs
            {
                WorkloadPool = $"{Pulumi.GoogleNative.Config.Project}.svc.id.goog"
            },
            NodePools = new InputList<NodePoolArgs>
            {
                new NodePoolArgs
                {
                    Name = "primary",
                    InitialNodeCount = config.RequireInt32("nodeCount"),
                    Config = new NodeConfigArgs
                    {
                        ImageType = "COS_CONTAINERD",
                        MachineType = config.Require("machineType"),
                        DiskType = "pd-ssd",
                        OauthScopes = 
                        {
                            "https://www.googleapis.com/auth/compute",
                            "https://www.googleapis.com/auth/devstorage.read_only",
                            "https://www.googleapis.com/auth/logging.write",
                            "https://www.googleapis.com/auth/monitoring",
                            "https://www.googleapis.com/auth/ndev.clouddns.readwrite"
                        },
                        GcfsConfig = new GcfsConfigArgs
                        {
                            Enabled = true
                        },
                        ShieldedInstanceConfig = new ShieldedInstanceConfigArgs
                        {
                            EnableIntegrityMonitoring = true,
                            EnableSecureBoot = true
                        }
                    },
                    Management = new NodeManagementArgs
                    {
                        AutoRepair = true,
                        AutoUpgrade = true
                    },
                    UpgradeSettings = new UpgradeSettingsArgs
                    {
                        MaxSurge = 1,
                        MaxUnavailable = 0
                    }
                }
            }
        });
```

**Comment by viveklak** [comment]
Looking again, this has to do with the mutations associated with clusters which don't use a patch and have explicit mutation operations. We started an initial look at this in https://github.com/pulumi/pulumi-google-native/pull/169 but we will want to revisit this.

**Comment by georgiosd** [comment]
Thanks! The only workaround is combining the old and new APIs together?

**Comment by viveklak** [comment]
Fixed in #588
